bffamily

```diff
@@ -29,22 +29,45 @@
 
 # Arrays are bash specific so this script requires bash.
 
-# map product family names to an appropriate regular expressions
-declare -A PRODUCT_FAMILY_MAP=(
-    ["Bluewhale"]="MBE(11|12)0(0|1)[A-F]-(BC|BN)[0-9]"
-    ["SmartNIC/Bekka"]="MBF1(L|M)332A-(AF|AE|AS|AT)(NAT|CAT)"
-    ["Lamina SmartNIC 100G"]="MBF1L516A-(ES|CS)(NAT|CAT)"
-    ["Tresnis"]="MBF1L516B-(ES|CS)(NAT|CAT)"
-    ["Sella BF1600"]="MBF1M6[A-F0-9]6A-(CS|CE)(NAT|CAT|NAN)"
-    ["Sella-SmartNIC/Follera"]="MBF1L666A-ES(CAT|NAT)"
-    ["Denarious"]="MBF1M216A-CSNAT_C16"
-    ["Poondion"]="MBF1M706A-CSNAT_C15"
-    ["Poondion BF1700"]="MBF1M716A-CSNAT"
-)
-
-BUS_NUMBER=$(lspci | grep -E "Mellanox.+MT416842\s+BlueField.+ConnectX-5" | grep -m 1 -oE "[0-9]{2}:[0-9]{2}.[0-9]")
-
-PART_NUMBER=$(lspci -s "$BUS_NUMBER" -vv | grep PN)
+BF1_PLATFORM_ID=0x00000211
+BF2_PLATFORM_ID=0x00000214
+
+bfversion=$(bfhcafw mcra 0xf0014.0:16)
+
+if [ "$bfversion" = $BF1_PLATFORM_ID ]; then
+    # map product family names to an appropriate regular expressions
+    declare -A PRODUCT_FAMILY_MAP=(
+        ["Bluewhale"]="MBE(11|12)0(0|1)[A-F]-(BC|BN)[0-9]"
+        ["SmartNIC/Bekka"]="MBF1(L|M)332A-(AF|AE|AS|AT)(NAT|CAT)"
+        ["Lamina SmartNIC 100G"]="MBF1L516A-(ES|CS)(NAT|CAT)"
+        ["Tresnis"]="MBF1L516B-(ES|CS)(NAT|CAT)"
+        ["Sella BF1600"]="MBF1M6[A-F0-9]6A-(CS|CE)(NAT|CAT|NAN)"
+        ["Sella-SmartNIC/Follera"]="MBF1L666A-ES(CAT|NAT)"
+        ["Denarious"]="MBF1M216A-CSNAT_C16"
+        ["Poondion"]="MBF1M706A-CSNAT_C15"
+        ["Poondion BF1700"]="MBF1M716A-CSNAT"
+    )
+    BUS_NUMBER=$(lspci | grep -E "Mellanox.+MT416842\s+BlueField.+ConnectX-5" | grep -m 1 -oE "[0-9]{2}:[0-9]{2}.[0-9]")
+    PART_NUMBER=$(lspci -s "$BUS_NUMBER" -vv | grep PN)
+
+elif [ "$bfversion" = $BF2_PLATFORM_ID ]; then
+
+    # map product family names to an appropriate regular expressions
+    declare -A PRODUCT_FAMILY_MAP=(
+        ["EVB"]="MBF2-EVB"
+        ["Camelot"]="MBF2(H|M)5(26A|16A|25A|15A)"
+        ["Atlantis"]="MBF2(H|M)(322A|332A)"
+        ["Arcadia OCP 2.0"]="MBF2M(922|912)A"
+        ["Asgard"]="MBF2H51(5|6)B"
+        ["BlueForce"]="BLUEFORCE_IPN"
+        ["BlueSphere"]="MBF2M512A"
```

Comment:
This is a different OPN than the one in the Wiki page (https://wikinox.mellanox.com/pages/viewpage.action?spaceKey=SW&title=BlueField2+Board+Part+Number+Mappings)

Reply:
I forgot to add BS2M512A but MBF2M512A is valid. no need to specify the end of it i.e. AENOT and AEEOT since they both refer to the Bluesphere family anyways.